browser-switch/src/test/java/com/braintreepayments/browserswitch/BrowserSwitchClientTest.java

```diff
@@ -0,0 +1,309 @@
+package com.braintreepayments.browserswitch;
+
+import android.content.Context;
+import android.content.Intent;
+import android.net.Uri;
+
+import androidx.fragment.app.FragmentActivity;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.mockito.ArgumentCaptor;
+import org.mockito.InOrder;
+import org.mockito.Mockito;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertNotNull;
+import static org.junit.Assert.assertNull;
+import static org.junit.Assert.assertSame;
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyInt;
+import static org.mockito.ArgumentMatchers.eq;
+import static org.mockito.ArgumentMatchers.isNull;
+import static org.mockito.ArgumentMatchers.same;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.never;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+public class BrowserSwitchClientTest {
+
+    private Uri uri;
+    private Intent browserSwitchIntent;
+    private FragmentActivity activity;
+
+    private BrowserSwitchListener listener;
+
+    private Context applicationContext;
+    private BrowserSwitchRequest request;
+
+    private ActivityFinder activityFinder;
+    private BrowserSwitchConfig browserSwitchConfig;
+
+    private BrowserSwitchPersistentStore persistentStore;
+
+    private BrowserSwitchClient sut;
+
+    @Before
+    public void beforeEach() {
+        persistentStore = mock(BrowserSwitchPersistentStore.class);
+
+        uri = mock(Uri.class);
+        browserSwitchIntent = mock(Intent.class);
+        activity = mock(FragmentActivity.class);
+
+        listener = mock(BrowserSwitchListener.class);
+
+        activityFinder = mock(ActivityFinder.class);
+        applicationContext = mock(Context.class);
+
+        browserSwitchConfig = mock(BrowserSwitchConfig.class);
+
+        request = mock(BrowserSwitchRequest.class);
+
+        // global scaffold
+        when(activity.getApplicationContext()).thenReturn(applicationContext);
+
+        sut = BrowserSwitchClient.newInstance(browserSwitchConfig, activityFinder, persistentStore);
+    }
+
+    @Test
+    public void start_whenRequestCodeIsIntegerMinValue_callsbackListenerWithError() {
```

Comment:
```suggestion
    public void start_whenRequestCodeIsInNegativeInteger_callsbackListenerWithError() {
```

Reply:
This one's interesting the OG implementation used `Integer.MIN_VALUE` almost like a special distinct value. I agree we probably will never have a negative request code, but it is legit checking that request code equals the minimum possible integer that can be represented on the device.

Reply:
Ah, ok - yeh I see now in the old code how we were setting the request code to `Integer.MIN_VALUE` in cases where there was a BrowserSwitch error or while BrowserSwitch was occurring.

Reply:
So this test is more of a "backwards compatibility" test.

Reply:
Yeah we may revise going forward. Trying to make the new code line up with the previous implementation will help with the review though I'm thinking.

The parts that need changing should stick out more than the rest as we move to refactor the other modules we can prioritize the behavioral changes.